### PR TITLE
Add OpenShift distro in cluster api

### PR DESCRIPTION
**Pull request by shanchunyang0919**
Support the following field in our `Cluster` object
```
status:
  distribution: OPENSHIFT
```

Testcases pass
```
go test .
ok      github.com/cisco-open/cluster-registry-controller/pkg/clustermeta       0.409s
```
## Example 
Screenshot 2023-02-06 at 2 20 43 PM

### CODE REVIEW

File: pkg/clustermeta/distributions.go [hunk 1]
```diff
@@ -230,3 +238,39 @@ func IsIKS(ctx context.Context, client client.Client, node *corev1.Node) (match
...
+	if err != nil {
+		return
+	}
+
+	// Note: We currently support ROSA setup where we install OpenShift on AWS clusters so the provider should be Amazon.
+	if provider != AMAZON {
+		match = false
+
+		return
+	}
```
Comment (Laci21) [inline pkg/clustermeta/distributions.go:273]: What if someone uses an OpenShift on prem cluster? Distribution still be "openshift" in that case, so I'd suggest to leave out this check.

Comment (shanchunyang0919) [reply pkg/clustermeta/distributions.go]: Thanks for the review @Laci21. Fixed.

**Comment by shanchunyang0919** [comment]
#### Update (02/09)
Screenshot 2023-02-09 at 4 09 23 AM

Test passed
```
go test .          
ok      github.com/cisco-open/cluster-registry-controller/pkg/clustermeta       0.693s
```

**Review by tiswanso — APPROVED** [review]
lgtm

**Comment by shanchunyang0919** [comment]
> lgtm

thanks for reviewing folks.

**Review by Laci21 — APPROVED** [review]
lgtm, thanks @shanchunyang0919!

**Comment by shanchunyang0919** [comment]
Thanks for review @Laci21

**Pull request merged**